test/parallel/test-runner-watch-coverage.mjs

```diff
@@ -0,0 +1,84 @@
+// Flags: --expose-internals
+import * as common from '../common/index.mjs';
+import { describe, it } from 'node:test';
+import { spawn } from 'node:child_process';
+import { writeFileSync } from 'node:fs';
+import util from 'internal/util';
+import tmpdir from '../common/tmpdir.js';
+import assert from 'node:assert';
+
+if (common.isIBMi)
+  common.skip('IBMi does not support `fs.watch()`');
+
+function getCoverageFixtureReport() {
+  const report = [
+    '# start of coverage report',
+    '# ---------------------------------------------------------------',
+    '# file           | line % | branch % | funcs % | uncovered lines',
+    '# ---------------------------------------------------------------',
+    '# dependency.js  | 100.00 |   100.00 |  100.00 | ',
+    '# dependency.mjs | 100.00 |   100.00 |  100.00 | ',
+    '# test.js        | 100.00 |   100.00 |  100.00 | ',
+    '# ---------------------------------------------------------------',
+    '# all files      | 100.00 |   100.00 |  100.00 |',
+    '# ---------------------------------------------------------------',
+    '# end of coverage report',
+  ].join('\n');
+
+  if (common.isWindows) {
+    return report.replaceAll('/', '\\');
+  }
+
+  return report;
+}
+
+tmpdir.refresh();
+
+const fixtureContent = {
+  'dependency.js': 'module.exports = {};',
+  'dependency.mjs': 'export const a = 1;',
+  'test.js': `
+const test = require('node:test');
+require('./dependency.js');
+import('./dependency.mjs');
+import('data:text/javascript,');
+test('test has ran');`,
+};
+
+const fixturePaths = Object.keys(fixtureContent)
+  .reduce((acc, file) => ({ ...acc, [file]: tmpdir.resolve(file) }), {});
+Object.entries(fixtureContent)
+  .forEach(([file, content]) => writeFileSync(fixturePaths[file], content));
```

Comment:
Please don't create the file sync

Reply:
?

Reply:
Sorry @rluvaton, but I didn't get your point 😞. Please explain a little bit.

Reply:
this is a test file, dosent seem to important to me @rluvaton

Reply:
I thought to avoid blocking other things and we can write all the files in parrallel